CFC_WebApp/main/zipcode.py

```diff
@@ -17,18 +16,26 @@ def getDistinctUserCount():
     distinctUserCount = len(Profiles.distinct("user_id"))
     logging.debug("Found %s distinct users " % distinctUserCount)
     return distinctUserCount
+   
 
-def get_userZipcode(user):
+def get_userZipcode(user, valid_zip):
     location = detect_home_from_db(user)
-    if location!='N/A':
+    current_db = get_profile_db()
+    user_pro = current_db.find_one({"$and":[{'source':'Shankari'},{'user_id':user}]})
```

Comment:
Would have been nice to use the User DAO, which didn't exist at the time this code was written, but was added later. That has a built-in method to get the profile:
User.from_uuid(user).getProfile()